src/Spin Button/main.vala

```diff
@@ -0,0 +1,40 @@
+#! /usr/bin/env -S vala workbench.vala --pkg libadwaita-1
+
+string tell_time (Gtk.SpinButton hours, Gtk.SpinButton minutes) {
+    return (@"The time selected is $(hours.text):$(minutes.text)");
+}
+
+public void main () {
+    var hours = (Gtk.SpinButton) workbench.builder.get_object ("hours");
```

Comment:
I'd have these as global variables rather than local and having to pass them as parameters.